shared/fs/container.js

```diff
@@ -11,39 +11,58 @@ type OwnProps = {
 }
 
 type StateProps = {
+  _itemNames: I.List<string>,
+  _username?: string,
+  _pathItems: I.Map<Types.Path, Types.PathItem>,
+  _sortSetting: Types.SortSetting,
+
   path: Types.Path,
-  items: I.List<string>,
   progress: Types.ProgressType,
 }
 
-type DispatchProps = {}
+type DispatchProps = {
+  loadFolderList: (path: Types.Path) => void,
+}
 
 const mapStateToProps = (state: TypedState, {routeProps}: OwnProps) => {
   const path = Types.stringToPath(routeProps.get('path', Constants.defaultPath))
   const itemDetail = state.fs.pathItems.get(path)
-  const items = itemDetail && itemDetail.type === 'folder' ? itemDetail.get('children', I.List()) : I.List()
-  const progress: Types.ProgressType = itemDetail ? itemDetail.progress : 'pending'
-  return {items, path, progress}
+  return {
+    _itemNames: itemDetail && itemDetail.type === 'folder' ? itemDetail.get('children', I.List()) : I.List(),
+    _username: state.config.username || undefined,
+    _pathItems: state.fs.pathItems,
+    _sortSetting: state.fs.pathUserSettings.get(path, Constants.makePathUserSetting()).get('sort'),
+    path,
+    progress: itemDetail ? itemDetail.progress : 'pending',
+  }
 }
 
 const mapDispatchToProps = (dispatch: Dispatch) => ({
-  _loadFolderList: (path: Types.Path) => dispatch(FsGen.createFolderListLoad({path})),
+  loadFolderList: (path: Types.Path) => dispatch(FsGen.createFolderListLoad({path})),
 })
 
-const mergeProps = ({path, items, progress}: StateProps, dispatchProps: DispatchProps, ownProps) => {
+const mergeProps = (stateProps: StateProps, dispatchProps: DispatchProps, ownProps) => {
+  const pathItems = stateProps._itemNames
+    .map(name => Types.pathConcat(stateProps.path, name))
+    .map(path => stateProps._pathItems.get(path, Constants.makeUnknownPathItem()))
+  const username = Types.pathIsNonTeamTLFList(stateProps.path) ? stateProps._username : undefined
+  const items = Constants.sortPathItems(pathItems, stateProps._sortSetting, username)
+    .map(({path}) => path)
+    .toArray()
   return {
-    items: items.map(name => Types.pathConcat(path, name)).toArray(),
-    path,
-    progress,
-    ...dispatchProps,
+    items,
```

Comment:
Maybe `items` here should actually be an array of `PathItem`s. That way we don't need to collapse it back to an array of paths. WDYT?

Reply:
Hmm `FileRow` still needs `elems` (derived from `path`) to determine if it's a team name or just regular item so we can't just replace it with an array of `PathItem`s. We could extend `PathItem` to include a `path` I guess, but not sure if that's actually better since we'd be duplicating the `path` everywhere.

Reply:
Ohhh I was confused about what I was reading.

All right, let's just leave it the way you've made it.